providers/cncf/kubernetes/tests/unit/cncf/kubernetes/test_pod_generator.py

```diff
@@ -167,6 +167,61 @@ def setup_method(self):
             ),
         )
 
+    def test_pod_spec_for_task_sdk_runs(self, data_file):
+        template_file = data_file("pods/generator_base_with_secrets.yaml").as_posix()
+        worker_config = PodGenerator.deserialize_model_file(template_file)
+        result = PodGenerator.construct_pod(
+            dag_id="dag_id",
+            task_id="task_id",
+            pod_id="pod_id",
+            kube_image="test-image",
+            try_number=3,
+            date=self.logical_date,
+            args=["command"],
+            pod_override_object=None,
+            base_worker_pod=worker_config,
+            namespace="namespace",
+            scheduler_job_id="uuid",
+            content_json_for_volume='{"token":"mock","ti":{"id":"4d828a62-a417-4936-a7a6-2b3fabacecab","task_id":"mock","dag_id":"mock","run_id":"mock","try_number":1,"map_index":-1,"pool_slots":1,"queue":"default","priority_weight":1},"dag_rel_path":"mock.py","bundle_info":{"name":"n/a","version":"no matter"},"log_path":"mock.log","kind":"ExecuteTask"}',
```

Comment:
Let's stick a single-quote in here somewhere to double check the escaping works?

Reply:
Will do

Reply:
Added a test in the latest commit